### PR TITLE
Update README file

### PR DESCRIPTION
**Pull request by linus-sherrill**
Update README file to RST from MD

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+
+
+The KWIVER toolkit is a collection of software tools designed to
+tackle challenging image and video analysis problems and other related
+challenges. Recently started by Kitware’s Computer Vision and
+Scientific Visualization teams, KWIVER is an ongoing effort to
+transition technology developed over multiple years to the open source
+domain to further research, collaboration, and product development.
+
+The project is structured with the parent `kwiver` repository working
```
Comment (mleotta) [inline README.rst]: This sentence is no longer true

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+Building KWIVER
+===============
+
+Fletch
+------
+
+KWIVER has (and will have more) a number of dependencies on 3rd party
+Open Source libraries.  To make it easier to build KWIVER, especially
+on systems like Microsoft Windows that don't have package manager,
+[Fletch](https://github.com/Kitware/fletch) was developed to gather,
```
Comment (mleotta) [inline README.rst]: Need to change from markdown style hyperlinks to restructured text throughout this document

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+configure and build those packages for use with KWIVER.  Fletch is a
+[CMake](www.cmake.org) based "super-build" that takes care of most of
+the build details for you.
+
+To build Fletch, clone the Fletch repository:
+
+	git clone https://github.com/Kitware/fletch.git
+
+	git submodule update --init
```
Comment (mleotta) [inline README.rst]: I don't think this is true anymore

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+
+The project is structured with the parent `kwiver` repository working
+as CMake "super-build" that pulls in a number of KWIVER and other open
+source projects.
+
+
+Building KWIVER
+===============
+
+Fletch
```
Comment (mleotta) [inline README.rst]: This whole section should be largely removed and replaced with a URL to the Fletch page on GIthub.  The only thing needed here are a couple of high-level comments about why we use Fletch and info about how your choice of Fletch packages impacts what you can build in KWIVER.

Any specific instructions about how to build Fletch should be moved to the Fletch readme if they are not already covered there.

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+* `-DPYTHON_LIBRARY=/path/to/python/library` For example, for a default Python 2.7 Anaconda install on Linux, this would be `${HOME}/anaconda/lib/libpython2.7.so`
+
+Once your `cmake` command has completed, you can build with the following command
+
+	make
+
+kwiver
+------
+
+Once Fletch has been built, it's possible to build the `kwiver` repository as well.  This repo is also a CMake super-build
```
Comment (mleotta) [inline README.rst]: not a super-build

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+and can be fetched with this command:
+
+	git clone https://github.com/Kitware/kwiver.git
```
Comment (mleotta) [inline README.rst]: probably obvious to someone looking at the Github page.

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,484 @@
...
+
+The build can be configured with this command:
+
+	cmake -DKWIVER_ENABLE_PYTHON:BOOL=ON -DKWIVER_ENABLE_ARROWS:BOOL=ON -DKWIVER_ENABLE_PROCESSES:BOOL=ON -DKWIVER_ENABLE_TESTS:BOOL=ON -DKWIVER_ENABLE_TOOLS:BOOL=ON -DKWIVER_USE_BUILD_TREE:BOOL=ON -DKWIVER_ENABLE_VXL:BOOL=ON -DKWIVER_ENABLE_SPROKIT:BOOL=ON -DKWIVER_ENABLE_OPENCV:BOOL=ON -Dfletch_DIR:PATH=/path/to/fletch/build/directory /path/to/kwiver/source/directory
+
+As with Fletch, if you want to specify a particular Python
```
Comment (mleotta) [inline README.rst]: Does KWIVER automatically reuse the same Python specified for Fletch?  If not, it probably should.

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+------
+
+Once Fletch has been built, it's possible to build the `kwiver` repository as well.  This repo is also a CMake super-build
+and can be fetched with this command:
+
+	git clone https://github.com/Kitware/kwiver.git
+
+The build can be configured with this command:
+
+	cmake -DKWIVER_ENABLE_PYTHON:BOOL=ON -DKWIVER_ENABLE_ARROWS:BOOL=ON -DKWIVER_ENABLE_PROCESSES:BOOL=ON -DKWIVER_ENABLE_TESTS:BOOL=ON -DKWIVER_ENABLE_TOOLS:BOOL=ON -DKWIVER_USE_BUILD_TREE:BOOL=ON -DKWIVER_ENABLE_VXL:BOOL=ON -DKWIVER_ENABLE_SPROKIT:BOOL=ON -DKWIVER_ENABLE_OPENCV:BOOL=ON -Dfletch_DIR:PATH=/path/to/fletch/build/directory /path/to/kwiver/source/directory
```
Comment (mleotta) [inline README.rst]: I would not suggest a specific set of build options here.  Rather talk about some of the high-level build options and what they do, and let the user decide.

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+Once you've built KWIVER, you'll want to test that it's working on
+your system.  From a command prompt execute the following command:
+
+	source </path/to/kwiver/build>/install/setup_KWIVER.sh
+
+Where `</path/to/kwiver/build>` is the actual path of your KWIVER CMake build directory.
+
+This will set up your PATH, PYTHONPATH and other environment variables to allow KWIVER to work conveniently.
+
+The central component of KWIVER is [Sprokit](www.sprokit.org).  We use
```
Comment (mleotta) [inline README.rst]: The central component of KWIVER is Vital, not Sprokit.  I don't even build Sprokit for MAP-Tk.

File: README.rst [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+Once your `cmake` command has completed, use `make` (on Linux) to build it.
+
+
+Running KWIVER
+==============
+
+Once you've built KWIVER, you'll want to test that it's working on
+your system.  From a command prompt execute the following command:
+
+	source </path/to/kwiver/build>/install/setup_KWIVER.sh
```
Comment (mleotta) [inline README.rst]: This works on Linux/Mac only, is there a Windows equivalent?

File: README.rst [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+
+Where `</path/to/kwiver/build>` is the actual path of your KWIVER CMake build directory.
+
+This will set up your PATH, PYTHONPATH and other environment variables to allow KWIVER to work conveniently.
+
+The central component of KWIVER is [Sprokit](www.sprokit.org).  We use
+Sprokit's pipelining facilities to manage, integrate and run many of
+KWIVER's modules and capabilities.  To see what modules (called
+processes in Sprocket) are available, run the following command::
```
Comment (mleotta) [inline README.rst]: Sprocket --> Sprokit

File: README.rst [hunk 12]
```diff
@@ -0,0 +1,484 @@
...
+
+    $ plugin_explorer --process -b
```
Comment (mleotta) [inline README.rst]: I think it's good to talk about `plugin_explorer` and different types of plugins in general at this level, but I would not put the entire output right in the main readme.  We could link to a separate document the walks through this example in more detail.

File: README.rst [hunk 13]
```diff
@@ -0,0 +1,484 @@
...
+what it says: it creates a Python thread for every process in the
+pipeline::
+
+	pipeline_runner -S pythread_per_process -p </path/to/kwiver/source>/sprokit/pipelines/number_flow_python.pipe>
+
+As with the previous pipeline, the numbers will be written to an output file, this time `numbers_from_python.txt`
+
+
+
+vital
```
Comment (mleotta) [inline README.rst]: We should talk about Vital before Sprokit

File: README.rst [hunk 14]
```diff
@@ -0,0 +1,484 @@
...
+=====
+
+Vital is an open source C++ collection of libraries and tools that
+supply basic types and services to the Kitware KWIVER imagery tool
+kit.
```
Comment (mleotta) [inline README.rst]: We should expand on this.  I can help with this if needed.

File: README.rst [hunk 15]
```diff
@@ -0,0 +1,484 @@
...
+Dependencies
+------------
+
+Vital has minimal required dependencies. Sprokit pipeline framework
+relies on boost.  Arrows and sprokit processes are structured so that
+the code that depends on an external package is in a directory with
+the major dependency name (e.g. vxl, ocv). The dependencies can be
+turned ON or OFF through CMake variables.
+
+Development
```
Comment (mleotta) [inline README.rst]: I would move this to a separate doc and link from here.  It's often called "CONTRIBUTING.rst".

File: README.rst [hunk 16]
```diff
@@ -0,0 +1,484 @@
...
+
+    $ ccmake /path/to/kwiver/source
+
+where the path above is the location of your kwiver source tree.  The
+ccmake tool allows for interactive selection of CMake options.
+Alternatively, using the CMake GUI you can set the source and build
+directories accordingly and press the "Configure" button.
+
+
+CMake Options
```
Comment (mleotta) [inline README.rst]: This list  is missing ENABLE_ARROWS.

There is no discussion of arrow in this whole readme.

**Comment by mleotta** [comment]
The conversion to RST is good, but the content needs a lot of work.

As a starting point, let's factor out the specific sprokit example into a separate doc.  Do the same for the section for developers.

The top level readme should just be an introduction to the project and a high level overview with information about where to find out more detail.
 👎

File: README.rst (hunk 7 above)
Comment (linus-sherrill) [inline README.rst]: Currently fletch does not export its PYTHON  version or other useful attributes.

File: README.rst (hunk 10 above)
Comment (linus-sherrill) [inline README.rst]: We need to create a setup_KWIVER.bat for windows

File: README.rst (hunk 14 above)
Comment (linus-sherrill) [inline README.rst]: Thanks, I could use the help

File: README.rst [hunk 17]
```diff
@@ -0,0 +1,492 @@
...
+
+Acknowledgements
+================
+
+The authors would like to thank AFRL/Sensors Directorate for their support
+of this work via SBIR Contract FA8650-14-C-1820. The portions of this work
+funded by the above contract are approved for public release via case number
+88ABW-2017-2725.
+
+The authors would like to thank IARPA for their support of this work via the
```
Comment (mleotta) [inline README.rst]: @collinsr please review this line.

File: README.rst [hunk 18]
```diff
@@ -0,0 +1,492 @@
...
+DIVA program.
+
+The authors would like to thank NOAA for their support of this work via the
```
Comment (mleotta) [inline README.rst]: @linus-sherrill @mattdawkins please review this line.

File: README.rst (hunk 14 above)
Comment (mleotta) [inline README.rst]: I've put the overview description of Vital at the top.  I think the remaining content in this section should move into an overview of Vital document in `doc/manuals/vital`

File: README.rst [hunk 19]
```diff
@@ -0,0 +1,502 @@
...
+
+	source </path/to/kwiver/build>/install/setup_KWIVER.sh
+
+Where `</path/to/kwiver/build>` is the actual path of your KWIVER
+CMake build directory.
+
+This will set up your PATH, PYTHONPATH and other environment variables
+to allow KWIVER to work conveniently.
+
+The central component of KWIVER is vital which supplies basic data
```
Comment (mleotta) [inline README.rst]: @linus-sherrill I think we should move the rest of this section (starting at this line) to a separate document in `doc/manuals/sprokit` that is called `getting-started.rst` or something similar.  In the remainder of this section I would like to see a much simplified "hello world" example of how to run plugin_explorer, mostly to confirm that everything was built and/or installed correctly.  It should not be specific to sprokit, but indicate how to list all arrows plugins and all sprokit plugins.

**Pull request merged**